pkg/artifactory/user.go

```diff
@@ -0,0 +1,211 @@
+package artifactory
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+type User struct {
+	Name                     string   `json:"name"`
+	Email                    string   `json:"email"`
+	Password                 string `json:"password,omitempty"`
+	Admin                    bool     `json:"admin"`
+	ProfileUpdatable         bool     `json:"profileUpdatable"`
+	DisableUIAccess          bool     `json:"disableUIAccess"`
+	InternalPasswordDisabled bool     `json:"internalPasswordDisabled"`
+	LastLoggedIn             string   `json:"lastLoggedIn"`
+	Realm                    string   `json:"realm"`
+	Groups                   []string `json:"groups"`
+}
+
+var baseUserSchema = map[string]*schema.Schema{
+	"name": {
+		Type:         schema.TypeString,
+		Required:     true,
+		ForceNew:     true,
+		ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
Do we need to use ValidateDiagFunc? instead of ValidateFunc?

Reply:
Good catch. Will update.